### PR TITLE
fix: Block Settings Sidebar refresh when we change the Editor responsive Preview.

### PR DESCRIPTION
**Pull request by RajkiranBagal**
### Description
Task - https://app.startinfinity.com/b/Mi5zu3VRVzk/3sAgbH1y7qY/fb66dbcc-b890-4676-8008-d47e65409f76?redirect=true&view=6e170ea7-8708-4546-8b82-23d2eebdf694&t=attributes

### Types of changes
 Bug fix (non-breaking change which fixes an issue)

### How has this been tested?
- Added tets cases in the task.

### Checklist:
- [x] My code is tested
- [x] I've included any necessary tests 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: blocks-config/uagb-controls/getUAGEditorStateLocalStorage.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const getUAGEditorStateLocalStorage = ( key = false ) => {
+	if ( ! window.localStorage ) {
+		return null;
+	}
+
+	if ( ! key ) {
+		return localStorage; // eslint-disable-line no-undef
+	}
+
+	const uagLastOpenedState = localStorage.getItem( 'uagLastOpenedState' ); // eslint-disable-line no-undef
```
Comment (sandeshjangam) [inline blocks-config/uagb-controls/getUAGEditorStateLocalStorage.js:10]: YOu can add this in eslint global `localStorage` No need to ignore eslint

File: src/components/responsive-toggle/index.js [hunk 2]
```diff
@@ -6,20 +6,21 @@
  import { __ } from '@wordpress/i18n'
  import { useState, useCallback } from '@wordpress/element'
  import { dispatch } from '@wordpress/data'
- 
+ import getUAGEditorStateLocalStorage from '@Controls/getUAGEditorStateLocalStorage';
```
Comment (sandeshjangam) [inline src/components/responsive-toggle/index.js]: Extra white space before all lines

```suggestion
import getUAGEditorStateLocalStorage from '@Controls/getUAGEditorStateLocalStorage';
```

File: src/blocks.js [hunk 3]
```diff
@@ -8,6 +8,13 @@
  * All blocks should be included here since this is the file that
  * Webpack is compiling as the input file.
  */
+
+ import getUAGEditorStateLocalStorage from '@Controls/getUAGEditorStateLocalStorage';
+
+// Delete the local storage on every refresh.
+const uagLocalStorage = getUAGEditorStateLocalStorage();
+uagLocalStorage.removeItem( 'uagLastOpenedState' ); // eslint-disable-line no-undef
```
Comment (sandeshjangam) [inline src/blocks.js]: what if this is `uagLocalStorage` is `null`  ?

File: blocks-config/uagb-controls/getUAGEditorStateLocalStorage.js (hunk 1 above)
Comment (RajkiranBagal) [reply blocks-config/uagb-controls/getUAGEditorStateLocalStorage.js]: Done

File: src/components/responsive-toggle/index.js (hunk 2 above)
Comment (RajkiranBagal) [reply src/components/responsive-toggle/index.js]: Removed

File: src/blocks.js (hunk 3 above)
Comment (RajkiranBagal) [reply src/blocks.js]: Checked conditions.

**Review by sandeshjangam — APPROVED** [review]

**Review by sandeshjangam — APPROVED** [review]

**Pull request merged**